tests/test_connector.py

```diff
@@ -1906,6 +2012,8 @@ def check_with_exc(err):
 
         assert not conn._waiters
 
+        yield from conn.close()
```

Comment:
I wonder if this function should just be converted to an async function? Maybe the test just predates async syntax...

The codecov comment says that it's not running though, which makes sense given all we are doing is creating a generator object. So, we never actually run any of the function body...

Reply:
That's an interesting idea… I'll have to check. I'm tired, working on this patch on and off for years. So at this point, I'm almost ready to accept any suggestion that would get this merged. I just want to be careful not to break anything. I think the last couple of times (like 6–7 years ago) I attempted integrating xdist here, and it brought a lot of flakiness that would become evident in Travis CI. So I'm both invested and annoyed at this point :)